### DISCUSSION TITLE
How digraph can achieve a similar function in has_path with graph?

### DISCUSSION THREAD

**Discussion by bucongfan** [networkx/networkx#6884] [Q&A] [answered]
If I build a digraph, How can I judge whether two leafnode have connected?

I don't want to use lowest_common_ancestor, because I found this method 10 times slower than only judging whether this path exists.

>>> G = nx.DiGraph()
>>> G.add_edges_from([(1, 2), (1, 3), (3, 4)])
>>> nx.has_path(G, 2, 4)
False
>>> nx.has_path(G, 2, 3)
False
>>> nx.has_path(G, 1, 3)
True
>>> G = nx.Graph()
>>> G.add_edges_from([(1, 2), (1, 3), (3, 4)])
>>> nx.has_path(G, 2, 4)
True

**Comment by MridulS** [comment]
You can keep an undirected copy of the directed Graph if you want to test if the nodes have a path between them (without caring about the direction).

```python
G = nx.DiGraph()
G.add_edges_from([(1, 2), (1, 3), (3, 4)])
H = G.to_undirected()
nx.has_path(H, 2, 4)
```

**Comment by bucongfan** [comment]
Thank you very much!

But I have a question, why not support to use digraph do the same thing? Is it not a common option?

**Reply by MridulS** [reply, marked as answer]
In a directed graph, there is no path in the other direction. There is a notion of weakly connected when finding components but I haven't really seen weak paths (where you just ignore the direction).

**Reply by bucongfan** [reply]
Thanks a lot!
